src/scripts/entities/princess.js

```diff
@@ -95,23 +95,22 @@ Princess.prototype.checkFuel = function () {
 function time() {
     return new Date().getTime();
 }
-var lastTime = time(), lastDirection = false;
 Princess.prototype.move = function move(direction) {
     // Set new facing direction
-    var data = this._data;
+    var data = this._data,
+        newSpeed = this.width * 6 + (this.game.turbo * this.game.turbo * 0.8);
     data.facing = direction;
-    // Modify this position
-    if (lastDirection != direction) {
-        lastTime = time();
-        lastDirection = direction;
-    }
-    if (!direction) {
-        return;
-    }
+    //if (!direction) {
+    //    return;
+    //}
+
     if (c.LEFT === direction) {
-        this.body.position.x -= c.STEP * 1.5;
+        this.body.velocity.x = -newSpeed;
```

Comment:
I think we need to set the speed taken from the touch-control:
```js
this.game.touchControl.speed;
```
Or, in the other case if we are using the keyboard, you can do something like:
```js
var speed = keyboard.down ? 30 : this.game.touchControl.speed; // I ignore which speed is the correct one
```
then, operating this speed with the current turbo value & stuff... hence, we'll have a variable velocity for the princess in function of how long is the dragging.